xc7/fasm2bels/bram_models.py

```diff
@@ -1060,17 +1060,27 @@ def process_bram(conn, top, tile, features):
     assert 'RAMB36.EN_ECC_WRITE' not in tile_features
 
     num_brams = 0
+    num_sdp_brams = 0
+    num_read_width_18 = 0
     for bram in brams:
         if 'IN_USE' in bram_features[bram]:
             num_brams += 1
+        # The following are counters to check whether the bram
+        # occupies both RAMB18 and is in SDP mode.
+        if 'SDP_READ_WIDTH_36' in bram_features[bram]:
+            num_sdp_brams += 1
+        if 'READ_WIDTH_A_18' in bram_features[bram]:
+            num_read_width_18 += 1
 
     assert num_brams >= 0 and num_brams <= 2, num_brams
 
+    is_bram_36 = num_sdp_brams == 2 and num_read_width_18 == 2
+
     sites = []
     for bram in sorted(brams):
         sites.append(process_bram_site(top, brams[bram], bram_features[bram]))
 
-    if num_brams == 2:
+    if num_brams == 2 and (num_sdp_brams == 0 or is_bram_36):
```

Comment:
Comment in this section needs an update explaining the relationship between the BRAM18's and the BRAM36